File: 2231-largest-number-after-digit-swaps-by-parity/2231.py
class Solution:
    def largestInteger(self, num: int) -> int:
        odd = []
        even = []
        array = [int(i) for i in str(num)]

        for a in array:
            if a % 2 == 0:
                heappush(even, -a)
            else:
                heappush(odd, -a)

        ret = 0
        for a in array:
            if a % 2 == 0:
                ret = ret * 10 - heappop(even)
            else:
                ret = ret * 10 - heappop(odd)

        return ret

File: 2231-largest-number-after-digit-swaps-by-parity/2231.cpp
class Solution {
public:
    int largestInteger(int num) {
        priority_queue<int> odd;
        priority_queue<int> even;
        vector<int> array;

        while (num > 0) {
            array.push_back(num % 10);
            num /= 10;
        }
        std::reverse(array.begin(), array.end());

        for (auto &n : array) {
            if ((n % 2) == 0) {
                even.push(n);
            } else {
                odd.push(n);
            }
        }

        int ret = 0;

        for (int i = 0; i < array.size(); ++i) {
            if ((array[i] % 2) == 0) {
                ret = ret * 10 + even.top();
                even.pop();
            } else {
                ret = ret * 10 + odd.top();
                odd.pop();
            }
        }

        return ret;
    }
};
